engine/src/main/java/org/terasology/engine/GameEngine.java

```diff
@@ -28,6 +28,23 @@
 public interface GameEngine extends AutoCloseable {
 
     /**
+     * @return The current, fine-grained status of the engine.
+     */
+    EngineStatus getStatus();
```

Comment:
Essentially what is the difference between being in a menu and being in game? There's no world and no player in the menu... although there is a player who is interacting with the menu, just no character they control. And having the world (or worlds) driven by entities is already in our todo. So in the end there are really only two states - loading (switching environment and everything that goes with that) and loaded (into a menu, or into a game). That is where I plan to head. It may be a twisty road.